src/mmrelay/matrix_utils.py

```
@@ -2614,7 +3161,54 @@ async def matrix_relay(
+            # These are expected initialization failures - log and retry
+            if init_attempt < max_init_retries - 1:
+                logger.warning(
+                    f"Matrix client initialization failed (attempt {init_attempt + 1}/{max_init_retries}): {exc}, "
+                    f"retrying in {init_retry_delay}s..."
+                )
+                await asyncio.sleep(init_retry_delay)
+            else:
+                logger.error(
+                    f"Matrix client initialization failed after {max_init_retries} attempts: {exc}. "
+                    f"Message to room {room_id} may be lost."
+                )
```

Ruff (0.14.14)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)